/*
 * This file was automatically generated by EvoSuite
 * Thu Oct 19 22:22:29 GMT 2023
 */

package org.apache.commons.net.ftp;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.net.ftp.FTPFileTest;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import org.opentest4j.MultipleFailuresError;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class FTPFileTest_ESTest extends FTPFileTest_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testHasPermissionTrue();
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testIsSymbolicLink();
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testHasPermissionInvalidFile();
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testGetTimestampInstantNullCalendar();
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      // Undeclared exception!
      try { 
        fTPFileTest0.testGetTimestampInstant();
        fail("Expecting exception: MultipleFailuresError");
      
      } catch(MultipleFailuresError e) {
         //
         // Multiple Failures (6 failures)\r
         // \torg.opentest4j.AssertionFailedError: expected: <2023> but was: <2014>\r
         // \torg.opentest4j.AssertionFailedError: expected: <AUGUST> but was: <FEBRUARY>\r
         // \torg.opentest4j.AssertionFailedError: expected: <4> but was: <14>\r
         // \torg.opentest4j.AssertionFailedError: expected: <23> but was: <20>\r
         // \torg.opentest4j.AssertionFailedError: expected: <40> but was: <21>\r
         // \torg.opentest4j.AssertionFailedError: expected: <55> but was: <21>
         //
         verifyException("org.junit.jupiter.api.AssertAll", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.toFormattedStringUnknownType();
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.toFormattedStringDirectoryType();
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testToString();
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testIsFile();
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testHasPermissionFalse();
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.toFormattedStringSymbolicLinkType();
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testIsUnknown();
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.toFormattedStringFileType();
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testIsDirectory();
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.toFormattedStringInvalidFile();
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      // Undeclared exception!
      try { 
        fTPFileTest0.toFormattedStringWithTimezone();
        fail("Expecting exception: MultipleFailuresError");
      
      } catch(MultipleFailuresError e) {
         //
         // Multiple Failures (1 failure)\r
         // \torg.opentest4j.AssertionFailedError: expected: <true> but was: <false>
         //
         verifyException("org.junit.jupiter.api.AssertAll", e);
      }
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      FTPFileTest fTPFileTest0 = new FTPFileTest();
      fTPFileTest0.testToStringDefault();
  }
}
